R/mod_common_forms.R

```diff
@@ -93,60 +93,236 @@ mod_common_forms_server <- function(
+          with(r$review_data, r$review_data[item_group == form, ]) |> 
+            dplyr::select(-dplyr::all_of(c("edit_date_time", "event_date"))), 
+          by = id_item
         ) |> 
-        create_table(expected_columns = names(form_items))
-      if(!input$show_all_data){ 
-        df <-  with(df, df[subject_id == r$subject_id, ]) 
+          dplyr::mutate(
+            item_value = ifelse(
+              reviewed == "No", 
+              paste0("<b>", htmltools::htmlEscape(item_value), "*</b>"), 
+              htmltools::htmlEscape(item_value)
+            )
+          ) |> 
+          create_table(expected_columns = names(form_items)) |> 
+          dplyr::mutate(o_reviewed = Map(\(x, y) append(x, list(row_id = y)), 
+                                         o_reviewed, 
+                                         dplyr::row_number()))
       }
-      df
+      common_form_data({
+        if(form == "Adverse events") {
+          df |> 
+            dplyr::filter(!grepl("Yes", `Serious Adverse Event`)
+            ) |> 
+            dplyr::select(-dplyr::starts_with("SAE"))
+        } else {
+          df
+        }
+      })
+      if (form == "Adverse events")
+        SAE_data({
+          df |>
+            dplyr::filter(grepl("Yes", `Serious Adverse Event`)) |>
+            dplyr::select(dplyr::any_of(
+              c("o_reviewed", "subject_id","form_repeat", "Name", "AESI",  "SAE Start date",
+                "SAE End date", "CTCAE severity", "Treatment related",
+                "Treatment action", "Other action", "SAE Category",
+                "SAE Awareness date", "SAE Date of death", "SAE Death reason")
+            )) |>
+            adjust_colnames("^SAE ")
+        })
     })
     
     mod_timeline_server("timeline_fig", r = r, form = form)
     
-    output[["SAE_table"]] <- DT::renderDT({
+    if (form == "Adverse events") {
+      output[["SAE_table"]] <- DT::renderDT({
+        req(form == "Adverse events")
+        datatable_custom(
+          isolate(subset(SAE_data(), input$show_all_data | subject_id == r$subject_id)), 
+          rename_vars = c("Review Status" = "o_reviewed", table_names), 
+          rownames= FALSE,
+          title = "Serious Adverse Events", 
+          escape = FALSE,
+          selection = "none",
+          callback = checkbox_callback,
+          options = list(
+            columnDefs = list(
+              list(
+                targets = "o_reviewed",
+                orderable = FALSE,
+                render = checkbox_render
+              ),
+              list(
+                targets = "subject_id",
+                visible = isolate(input$show_all_data)
+              )),
+            rowCallback = row_callback
+          ))
+      })
+      SAE_proxy <- DT::dataTableProxy("SAE_table")
+    }
+    
+    output[["common_form_table"]] <- DT::renderDT({
+      datatable_custom(
+        isolate(subset(common_form_data(), input$show_all_data | subject_id == r$subject_id)), 
```

Comment:
`isolate` here is needed in order to prevent some refreshing loop/because the data is updated using the `dataTableProxy`, right?

Reply:
Yes, `isolate()` causes the render to only run once.